src/com/mendix/widget/StarRating/components/StarRating.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, createElement, DOM } from "react";
+
+import * as Rating from "react-rating";
+import { Alert } from "./Alert";
+
+export interface StarRatingProps {
+    campaignEntity: string;
+    contextObject?: mendix.lib.MxObject;
+    initialRate?: number;
+    onChangeMicroflow?: string;
+    readOnly: boolean;
+    rateType: "single" | "average";
+    rateEntity: string;
+    rateAttribute: string;
+    averageAttribute: string;
+}
+
+export interface StarRatingState {
+    errorMessage: string;
+}
+
+export class StarRating extends Component<StarRatingProps, StarRatingState> {
+    private fractions = 1;
+    private contextObject: mendix.lib.MxObject | undefined;
```

Comment:
Reverted back to previous implementation, which was adhering to this convention.